src/package.ts

```diff
@@ -0,0 +1,57 @@
+import { privateEncrypt } from 'crypto';
+import contextKeys from './contextKeys';
+import { exec, pathExists } from './utilities';
+
+export class SPMPackage {
+	private constructor(
+        readonly folder: string,
+        public contents: any
+    ) {
+        this.setContextKeys()
+    }
+
+    public static async create(folder: string): Promise<SPMPackage> {
+        try {
+            let contents = await SPMPackage.loadPackage(folder)
+            return new SPMPackage(folder, contents)
+        } catch(error) {
+            // TODO: output errors
+            return new SPMPackage(folder, null)
+        }
+    }
+
+    public static async loadPackage(folder: string): Promise<any> {
+        const { stdout } = await exec('swift package describe --type json', { cwd: folder });
+        return JSON.parse(stdout)
+    }
+
+    public async reload() {
+        try {
+            this.contents = await SPMPackage.loadPackage(this.folder)
+        } catch {
+            
+        }
+        this.setContextKeys()
+    }
+
+    get products(): Array<any> {
+        return this.contents.products
+    }
+
+    get dependencies(): Array<any> {
+        return this.contents.dependencies
+    }
+
+    get targets(): Array<any> {
+        return this.contents.targets
+    }
+
+    private setContextKeys() {
+        if (this.contents == null) {
```

Comment:
This can probably just be `if (this.contents)`. That should check for both `null` and `undefined`. I assume `== null` does the same thing, but I've learned that using `==` in JavaScript is almost always a bad idea 😅

Reply:
Now that I'm adding interfaces it can only be `undefined` or a valid value.

Reply:
This is now a test for `undefined` and also if `===` now ESLint is telling me about all of these